tests/baselines/reference/dynamicNamesErrors.errors.txt

```diff
@@ -9,17 +7,13 @@ dynamicNamesErrors.ts(25,1): error TS2322: Type 'T1' is not assignable to type '
     Type 'number' is not assignable to type 'string'.
 
 
-==== dynamicNamesErrors.ts (5 errors) ====
+==== dynamicNamesErrors.ts (3 errors) ====
     const c0 = "1";
     const c1 = 1;
     
     interface T0 {
         [c0]: number;
-        ~~~~
-!!! error TS2718: Duplicate property '1'.
         1: number;
-        ~
-!!! error TS2733: Property '1' was also declared here.

```

Comment:
I was going to say that this sure felt wrong, but your playground link does show that doing effectively this in other forms does not error. (Makes me wonder if we're missing errors; but that's more of a design question I don't have context for.)

Reply:
I don't think we should allow it, but it's definitely a separate question -- and disallowing it will probably break some amount of code.